pilot/pkg/networking/core/v1alpha3/httproute.go

```diff
@@ -181,9 +180,27 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundHTTPRouteConfig(node *
 
 	util.SortVirtualHosts(virtualHosts)
 
-	if features.EnableFallthroughRoute.Get() && !useSniffing {
+	if !useSniffing {
 		// This needs to be the last virtual host, as routes are evaluated in order.
 		if util.IsAllowAnyOutbound(node) {
+			egressCluster := util.PassthroughCluster
+			// no need to check for nil value as the previous if check has checked
+			if node.SidecarScope.OutboundTrafficPolicy.EgressProxy != nil {
+				// user has provided an explicit destination for all the unknown traffic.
+				// build a cluster out of this destination
+				serviceFound := false
+				for _, service := range push.Services(node) {
+					if string(service.Hostname) == node.SidecarScope.OutboundTrafficPolicy.EgressProxy.Host {
+						serviceFound = true
+						egressCluster = istio_route.GetDestinationCluster(node.SidecarScope.OutboundTrafficPolicy.EgressProxy,
+							service, listenerPort)
+						break
+					}
+				}
+				if !serviceFound {
```

Comment:
This is a misconfiguration, right? We probably don't want to just silently do nothing? Should we explicitly blackhole this (502 response, like below)

Reply:
I dont think we should blackhole because if this service comes online, we will have to find a way to update the sidecars. I think we should still construct the cluster just like the way we do in virtual services. This way even if the cluster comes back online after sometime, we should be good to go. Given that point,  I think we can simplify the entire logic as follows

```
egressCluster = istio_route.GetDestinationCluster(node.SidecarScope.OutboundTrafficPolicy.EgressProxy, nil, listenerPort)
```

We dont even need to look up the egress gateway service in the code. This will prevent the listeners from being updated as well.